packages/usa-in-page-navigation/src/test/test-patterns/test-hidden-headers.twig

```diff
@@ -0,0 +1,45 @@
+{% macro inpagenav(items, base_class) %}
+
+  {% set list_item_class = 'usa-in-page-nav__item' %}
+  {% set current_class = 'usa-current' %}
+
+  <aside class="usa-in-page-nav"
+    data-title-text="On this page"
+    data-title-heading-level="h4"
+    data-scroll-offset="0"
+    data-root-margin="0px 0px 0px 0px"
+    data-threshold="1"
+  ></aside>
+
+  <main class="usa-prose">
+    <div
+        class="usa-summary-box"
+        role="region"
+        aria-labelledby="summary-box-key-information"
+    >
+        <div class="usa-summary-box__body">
+            <h1 class="usa-summary-box__heading" id="summary-box-key-information">
+                Test if in-page nav excludes hidden headers
+            </h1>
+            <div class="usa-summary-box__text">
+                <p>The in-page nav link list should show only two "This heading is visible" headers.</p>
+            </div>
+        </div>
+    </div>
+
+    <!-- This header exists outside of the class defined in data-main-content-selector -->
+    <h2>This heading is visible</h2>
+    <p>Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Morbi commodo, ipsum sed pharetra gravida, orci magna rhoncus neque, id pulvinar odio lorem non turpis. </p>
+    <h3>This heading is visible</h3>
+    <p>Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Morbi commodo, ipsum sed pharetra gravida, orci magna rhoncus neque, id pulvinar odio lorem non turpis. </p>
+    <h2 style="display:none">This heading is hidden with display:none</h2>
+    <h2 style="visibility:hidden">This heading is hidden with visibility:hidden</h2>
```

Comment:
## Optional

To improve clarity you _could_ add some styles to the hidden elements. Example below.

**Before**
![image](https://github.com/uswds/uswds/assets/3385219/cb12ff62-5fe5-4dd1-8539-8d11f12a267f)

**After**
<img width="1281" alt="image" src="https://github.com/uswds/uswds/assets/3385219/3a00c461-9c6c-49e0-8633-397328f23da8">

Here's the markup I used:

```html
  <div class="outline-2px outline-magenta padding-1" style="outline-style: dashed;">
    <pre>Display: none;</pre>
    <h2 style="display:none">This heading is hidden with display:none</h2>
  </div>
  <div class="outline-2px outline-magenta padding-1" style="outline-style: dashed;">
    <pre>Visibility: hidden;</pre>
    <h2 style="visibility:hidden">This heading is hidden with visibility:hidden</h2>
  </div>
  <div class="outline-2px outline-magenta padding-1" style="outline-style: dashed;">
    <pre>Opacity: 0;</pre>
    <h2 style="opacity: 0;">This heading is hidden with opacity:0;</h2>
  </div>
```

Reply:
Love this idea. Added in [a420068](https://github.com/uswds/uswds/pull/5393/commits/a420068f2e1791ea9b143a1b4c84cda0b91edc89).